### PR TITLE
HDDS-5928. Use the new PipelineChoosePolicy in Ozon…

### PR DESCRIPTION
**Pull request by guohao-rosicky**
Use the new PipelineChoosePolicy in Ozone Streaming

jira: https://issues.apache.org/jira/browse/HDDS-5928

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/choose/algorithms/PipelineChoosePolicyFactory.java [hunk 1]
```diff
@@ -45,6 +46,18 @@ private PipelineChoosePolicyFactory() {
   }
 
   public static PipelineChoosePolicy getPolicy(
+      ConfigurationSource conf, NodeStateManager nodeStateManager)
+      throws SCMException {
+    ScmConfig scmConfig = conf.getObject(ScmConfig.class);
+    if (scmConfig.getPipelineChoosePolicyName()
+        .equals(CapacityPipelineChoosePolicy.class.getName())) {
+      return new CapacityPipelineChoosePolicy(nodeStateManager);
```
Comment (szetszwo) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/choose/algorithms/PipelineChoosePolicyFactory.java:54]: Pass the conf.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java [hunk 2]
```diff
@@ -163,6 +163,8 @@
   private Predicate<LayoutVersionProto> layoutMatchCondition;
   private Predicate<LayoutVersionProto> layoutMisMatchCondition;
 
+  private final ConfigurationSource configuration;
```
Comment (szetszwo) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java:166]: Don't add this field.  Pass the conf in the CapacityPipelineChoosePolicy constructor.

**Review by szetszwo — COMMENTED** [review]
@guohao-rosicky , thanks for working on this.

- First of all, this is not specific to Ozone-Streaming so that we should move this out from HDDS-4454 and merge this to the master branch directly.
- Please describe the details (How it chooses new pipeline?  What are the rules?) of the new CapacityPipelineChoosePolicy and provide a few examples in the JIRA.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java (hunk 2 above)
Comment (guohao-rosicky) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java:166]: Thank you. I'll modify it and update JIRA.

**Comment by guohao-rosicky** [comment]
> @guohao-rosicky , thanks for working on this.
> 
> * First of all, this is not specific to Ozone-Streaming so that we should move this out from [HDDS-4454](https://issues.apache.org/jira/browse/HDDS-4454) and merge this to the master branch directly.
> * Please describe the details (How it chooses new pipeline?  What are the rules?) of the new CapacityPipelineChoosePolicy and provide a few examples in the JIRA.

Okay, how do I transfer jira out? Do I have to resubmit it?

**Comment by szetszwo** [comment]
@guohao-rosicky ,I just moved HDDS-5928 out from HDDS-4454.  Please remove "[Ozone-Streaming]" from the title of this pull request.  Thanks.

**Comment by szetszwo** [comment]
Actually, you may need to create another pull request for merging to the master branch but not the HDDS-4454 branch.

**Comment by guohao-rosicky** [comment]
> Actually, you may need to create another pull request for merging to the master branch but not the [HDDS-4454](https://issues.apache.org/jira/browse/HDDS-4454) branch.

It has been modified

**Comment by adoroszlai** [comment]
> Actually, you may need to create another pull request for merging to the master branch but not the [HDDS-4454](https://issues.apache.org/jira/browse/HDDS-4454) branch.

Closing this PR, since it needs different target branch.  @guohao-rosicky Please recreate PR against `master`.  Please let us know if you need help with that.

**Pull request closed without merging**